packages/models/src/lib/report.ts

```diff
@@ -1,20 +1,108 @@
 import { z } from 'zod';
-import { hasMissingStrings } from './implementation/utils';
 import {
-  PluginConfig,
-  RunnerOutput,
+  errorItems,
+  hasDuplicateStrings,
+  hasMissingStrings,
+} from './implementation/utils';
+import {
+  AuditMetadata,
   auditMetadataSchema,
-  auditResultSchema,
+  PluginConfig,
   pluginMetadataSchema,
 } from './plugin-config';
+import {
+  positiveIntSchema,
+  slugSchema,
+  unixFilePathSchema,
+} from './implementation/schemas';
 
 export type PluginOutput = RunnerOutput & {
   slug: string;
   date: string;
   duration: number;
 };
 
-export const auditReportSchema = auditMetadataSchema.merge(auditResultSchema);
+const sourceFileLocationSchema = z.object(
+  {
+    file: unixFilePathSchema('Relative path to source file in Git repo'),
+    position: z
+      .object(
+        {
+          startLine: positiveIntSchema('Start line'),
+          startColumn: positiveIntSchema('Start column').optional(),
+          endLine: positiveIntSchema('End line').optional(),
+          endColumn: positiveIntSchema('End column').optional(),
+        },
+        { description: 'Location in file' },
+      )
+      .optional(),
+  },
+  { description: 'Source file location' },
+);
+
+/**
+ * Define Zod schema for the Issue type.
+ */
+export const issueSchema = z.object(
+  {
+    message: z.string({ description: 'Descriptive error message' }).max(128),
+    severity: z.enum(['info', 'warning', 'error'], {
+      description: 'Severity level',
+    }),
+    // "Reference to source code"
+    source: sourceFileLocationSchema.optional(),
+  },
+  { description: 'Issue information' },
+);
+export type Issue = z.infer<typeof issueSchema>;
+/**
+ * Define Zod schema for the Audit type.
+ */
+export const auditResultSchema = z.object(
+  {
+    slug: slugSchema('References audit metadata'),
+    displayValue: z
+      .string({ description: "Formatted value (e.g. '0.9 s', '2.1 MB')" })
+      .optional(),
+    value: positiveIntSchema('Raw numeric value'),
+    score: z
+      .number({
+        description: 'Value between 0 and 1',
+      })
+      .min(0)
+      .max(1),
+    details: z
+      .object(
+        {
+          issues: z.array(issueSchema, { description: 'List of findings' }),
+        },
+        { description: 'Detailed information' },
+      )
+      .optional(),
+  },
+  { description: 'Audit information' },
+);
+
+/**
+ * Define Zod schema for the RunnerOutput type.
+ */
+export const runnerOutputSchema = z.object(
+  {
+    audits: z
+      .array(auditResultSchema, { description: 'List of audits' })
+      // audit slugs are unique
+      .refine(
+        audits => !getDuplicateSlugsInAudits(audits as never), //@TODO use hackfix types instead of never same as in plugin config
```

Comment:
Would be great to get rid of these `never`s :pray:

Reply:
done